### PR TITLE
[BugFix] fix issues in export sink operator

### PR DESCRIPTION
**Pull request by silverbullet233**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #16568

## Problem Summary(Required) ：

fix some issues in export sink operator
1.  #16568

`SinkIOBuffer:: execute_io_task` is called in the async pipeline_sink_io threads. This function will access the `SinkIOBuffer` held by related sink operator. We need to ensure that all chunks are processed before closing sink operator. So I fixed all the places where `SinkIOBuffer` is used to solve this potential problem.

2. memory statistics problem

The chunks we put in the execution queue are allocated in our own threads, and their memory usage will be recorded in the mem tracker, but the release of the chunks is done internally in the execution queue, and we have no chance to update the mem tracker. So I made the following changes to actively release the memory after processing the chunk to ensure that the memory statistics are accurate.
image

3. In the previous implementation, query_timeout was mistakenly used as the rpc timeout of BrokerFileSystem, which is unreasonable. I changed it to the default timeout(10s) in BrokerFileSystem.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by satanson — DISMISSED** [review]

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Review by stdpain — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_admit_test

File: be/src/exec/pipeline/sink/export_sink_operator.cpp [hunk 1]
```diff
@@ -81,40 +82,41 @@ void ExportSinkIOBuffer::close(RuntimeState* state) {
     SinkIOBuffer::close(state);
 }
 
-void ExportSinkIOBuffer::_process_chunk(bthread::TaskIterator<const ChunkPtr>& iter) {
+void ExportSinkIOBuffer::_process_chunk(bthread::TaskIterator<ChunkPtr>& iter) {
     --_num_pending_chunks;
     if (_is_finished) {
         return;
     }
```
Comment (fzhedu) [inline be/src/exec/pipeline/sink/export_sink_operator.cpp:89]: `--_num_pending_chunks;` should be updated after really done?

File: be/src/exec/pipeline/sink/mysql_table_sink_operator.cpp [hunk 2]
```diff
@@ -78,20 +78,22 @@ void MysqlTableSinkIOBuffer::close(RuntimeState* state) {
     SinkIOBuffer::close(state);
 }
 
-void MysqlTableSinkIOBuffer::_process_chunk(bthread::TaskIterator<const ChunkPtr>& iter) {
+void MysqlTableSinkIOBuffer::_process_chunk(bthread::TaskIterator<ChunkPtr>& iter) {
     --_num_pending_chunks;
     if (_is_finished) {
```
Comment (fzhedu) [inline be/src/exec/pipeline/sink/mysql_table_sink_operator.cpp:83]: diit

File: be/src/exec/pipeline/sink/sink_io_buffer.h [hunk 3]
```diff
@@ -79,14 +79,9 @@ class SinkIOBuffer {
         return Status::OK();
     }
 
-    virtual bool is_finished() { return _is_finished; }
+    virtual bool is_finished() { return _is_finished && _num_pending_chunks == 0; }
```
Comment (fzhedu) [inline be/src/exec/pipeline/sink/sink_io_buffer.h:82]: _num_pending_chunks == 0 can be reached on every error or normal cases?

**Review by murphyatwork — APPROVED** [review]

File: be/src/exec/pipeline/sink/sink_io_buffer.h [hunk 4]
```diff
@@ -107,18 +102,20 @@ class SinkIOBuffer {
         return _io_status;
     }
 
-    static int execute_io_task(void* meta, bthread::TaskIterator<const ChunkPtr>& iter) {
+    static int execute_io_task(void* meta, bthread::TaskIterator<ChunkPtr>& iter) {
         auto* sink_io_buffer = static_cast<SinkIOBuffer*>(meta);
+        SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(sink_io_buffer->_state->query_mem_tracker_ptr().get());
         for (; iter; ++iter) {
             sink_io_buffer->_process_chunk(iter);
+            (*iter).reset();
```
Comment (fzhedu) [inline be/src/exec/pipeline/sink/sink_io_buffer.h:110]: can `reset` on _process_chunk is returned in error status?

File: be/src/exec/pipeline/sink/export_sink_operator.cpp (hunk 1 above)
Comment (silverbullet233) [reply be/src/exec/pipeline/sink/export_sink_operator.cpp:89]: > `--_num_pending_chunks;` should be updated after really done?

This variable is introduced to facilitate limiting the memory usage of sink operator by limiting the number of chunks pending to be processed. I think it's not a very strict limitation here, it's ok to be updated before or after it's really done.

File: be/src/exec/pipeline/sink/sink_io_buffer.h (hunk 3 above)
Comment (silverbullet233) [reply be/src/exec/pipeline/sink/sink_io_buffer.h:82]: > _num_pending_chunks == 0 can be reached on every error or normal cases?

yes, in both normal and abnormal cases, all chunks must be processed once. The difference between them is that when an error occurs, cancel will be triggered. at this time, `_process_chunk` will only update `_num_pending_chunks` and will not really process chunk.

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**